market/ipni/ipni-provider/ipni-provider.go

```diff
@@ -491,59 +489,120 @@ func (p *Provider) handleGet(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "invalid CID: "+req, http.StatusBadRequest)
 			return
 		}
-		ad, err := p.getAd(r.Context(), b, providerID)
-		if err != nil {
-			if errors.Is(err, ErrNotFound) {
-				// Check if this is an entry CID
-				entry, err := p.getEntry(b, providerID)
-				if err != nil {
-					if errors.Is(err, ErrNotFound) {
-						log.Debugw("No Content Found", "CID", b.String())
-						http.Error(w, "", http.StatusNoContent)
-						return
-					}
-					log.Errorf("failed to get entry %s for peer %s: %w", b.String(), providerID, err)
-					http.Error(w, "", http.StatusInternalServerError)
+
+		h := r.Header.Get(ipnisync.CidSchemaHeader)
+
+		switch h {
+		case ipnisync.CidSchemaAdvertisement:
+			ad, err := p.getAd(r.Context(), b, providerID)
```

Comment:
This switch would be a lot cleaner if p.getAd returned serialized bytes

Reply:
If I don't return the actual ad then I cannot use the func for getHead() as head needs to be a signed advertisement. Either I add more code and process getHead() separately or just return the advertisement like now.

Reply:
Makes sense, I guess then we could just do it as a separate `getAdBytes` function for use here and keep `getAd` for head getting as is.

Either case no strong opinion, can also keep this in the current shape, it works too.

Reply:
Add the the bytes function. It makes get() easier.